api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -964,6 +964,11 @@ type AzureNodePoolPlatform struct {
 	// If not specified then Boot diagnostics will be disabled.
 	// +optional
 	Diagnostics *Diagnostics `json:"diagnostics,omitempty"`
+
+	// MachineIdentityID is used as the user-assigned identity to be assigned to the VMs
+	//
+	// +optional
+	MachineIdentityID string `json:"machineIdentityID,omitempty"`
```

Comment:
Could we document what is the behavior if `MachineIdentityID` is not provided?

Could we document if the resource associated to the ID needs to preexist beforehand in Azure?

Thanks

Reply:
Yeah I think we have discussed what happens if it is not provided a few times. 

I will add it explicitly here though, but this field could change as workload identity/MSI work progresses.

Reply:
> Yeah I think we have discussed what happens if it is not provided a few times.

Yes, I am aware :+1: . It is more for consumers of the API and other people that might not be aware of it.